backend/cn/core_to_mucore.mli

```diff
@@ -0,0 +1,426 @@
+module CAE = Cerb_frontend.Cabs_to_ail_effect
+module BT = BaseTypes
+module SBT = SurfaceBaseTypes
+module Mu = Mucore
+module Loc = Locations
+module C = Compile
+module IT = IndexTerms
+
+module SymSet : Set.S with type elt = Sym.t
+module SymMap : Map.S with type key = Sym.t
+module IdMap : Map.S with type key = Id.t
+
+(* TODO: BCP: Needs cleaning up here -- In the .ml file this is just:
+            module Pmap = struct
+              include Pmap
+              let filter_map compare f map =
+                Pmap.fold (fun key value acc ->
+                    match f key value with
+                    | Some value' -> Pmap.add key value' acc
+                    | None -> acc
+                  ) map (Pmap.empty compare)
+            end
+*)
+module Pmap :
+  sig
+    type ('key, 'a) map = ('key, 'a) Pmap.map
+    val empty : ('key -> 'key -> int) -> ('key, 'a) map
+    val is_empty : ('key, 'a) map -> bool
+    val mem : 'key -> ('key, 'a) map -> bool
+    val add : 'key -> 'a -> ('key, 'a) map -> ('key, 'a) map
+    val singleton : ('key -> 'key -> int) -> 'key -> 'a -> ('key, 'a) map
+    val remove : 'key -> ('key, 'a) map -> ('key, 'a) map
+    val merge :
+      ('key -> 'a option -> 'b option -> 'c option) ->
+      ('key, 'a) map -> ('key, 'b) map -> ('key, 'c) map
+    val union : ('key, 'a) map -> ('key, 'a) map -> ('key, 'a) map
+    val compare :
+      ('a -> 'a -> int) -> ('key, 'a) map -> ('key, 'a) map -> int
+    val equal :
+      ('a -> 'a -> bool) -> ('key, 'a) map -> ('key, 'a) map -> bool
+    val iter : ('key -> 'a -> unit) -> ('key, 'a) map -> unit
+    val fold : ('key -> 'a -> 'b -> 'b) -> ('key, 'a) map -> 'b -> 'b
+    val for_all : ('key -> 'a -> bool) -> ('key, 'a) map -> bool
+    val exist : ('key -> 'a -> bool) -> ('key, 'a) map -> bool
+    val filter : ('key -> 'a -> bool) -> ('key, 'a) map -> ('key, 'a) map
+    val partition :
+      ('key -> 'a -> bool) ->
+      ('key, 'a) map -> ('key, 'a) map * ('key, 'a) map
+    val cardinal : ('key, 'a) map -> int
+    val bindings_list : ('key, 'a) map -> ('key * 'a) list
+    val bindings :
+      ('key * 'a -> 'key * 'a -> int) ->
+      ('key, 'a) map -> ('key * 'a) Pset.set
+    val domain : ('key, 'a) map -> 'key Pset.set
+    val range : ('a -> 'a -> int) -> ('key, 'a) map -> 'a Pset.set
+    val min_binding : ('key, 'a) map -> 'key * 'a
+    val max_binding : ('key, 'a) map -> 'key * 'a
+    val choose : ('key, 'a) map -> 'key * 'a
+    val split :
+      'key -> ('key, 'a) map -> ('key, 'a) map * 'a option * ('key, 'a) map
+    val find : 'key -> ('key, 'a) map -> 'a
+    val lookup : 'key -> ('key, 'a) map -> 'a option
+    val map : ('a -> 'b) -> ('key, 'a) map -> ('key, 'b) map
+    val mapi : ('key -> 'a -> 'b) -> ('key, 'a) map -> ('key, 'b) map
+    val from_set : ('key -> 'v) -> 'key Pset.set -> ('key, 'v) map
+    val filter_map :
+      ('a -> 'a -> int) ->
+      ('a -> 'b -> 'c option) -> ('a, 'b) Pmap.map -> ('a, 'c) Pmap.map
+  end
```

Comment:
Sigh. [This is from Lem.](https://github.com/rems-project/lem/blob/master/ocaml-lib/pmap.ml). It's an old version of the OCaml Map module, before functorisation.

I don't know how much of this comes from Cerberus frontend, though I do think existing CN code should be transitioned to normal Stdlib Maps
```
13:49 ➜  cerberus git:(cn-tidying) ✗ grep "Pmap.empty" -R backend/cn
backend/cn/effectful.ml:        ) m (Pmap.empty cmp)
backend/cn/core_to_mucore.ml:      ) map (Pmap.empty compare)
backend/cn/core_to_mucore.ml:    fmap (Pmap.empty Sym.compare, [], false)
backend/cn/core_to_mucore.ml:          (Pmap.empty Int.compare,
backend/cn/core_to_mucore.ml:         ([], Pmap.empty Int.compare)
backend/cn/core_to_mucore.ml:    (return (Pmap.empty Sym.compare))
backend/cn/cLogicalFuns.ml:  let global_context = {label_defs = Pmap.empty Sym.compare; c_fun_pred_map; mu_call_funinfo} in
backend/cn/core_to_mucore.mli:                  ) map (Pmap.empty compare)
```